### PR TITLE
Allow Null in Relation Reference Search Widget

### PR DESCRIPTION
**Pull request by signedav**
Having NULL available in the combo box on a field filter, getting back all features containing no FK.

Same in the _Select / Filter Feature Form (Ctrl+F)_ - means a NULL + Equal To is like the Flag "IS NULL".

Additionally in the _Select / Filter Feature Form (Ctrl+F)_ the "Equal To" flag is chosen as default as soon as the value changed (like in the Value Relation Widget) like suggested here  #37843

### CODE REVIEW

**Comment by signedav** [comment]
Hi @nyalldawson 
Are there any objections in supporting the null in the search? Otherwise is it okay for you to merge this?
Thanks a lot.

**Comment by nyalldawson** [comment]
@signedav 
> When you introduced it, you passed false for the AllowNULL value here

From memory, getting this to work with relation widget was out of scope at the time and it was easier just to disable it then deal with the bunch of required improvements which would allow null handling for the widget which were missing at the time.

If you've tested and this works OK, then let's merge!

File: src/gui/editorwidgets/qgsrelationreferencesearchwidgetwrapper.cpp [hunk 1]
```diff
@@ -96,9 +101,17 @@ QString QgsRelationReferenceSearchWidgetWrapper::createExpression( QgsSearchWidg
     case QVariant::ULongLong:
     {
       if ( flags & EqualTo )
+      {
+        if ( v.isNull() )
```
Comment (nyalldawson) [inline src/gui/editorwidgets/qgsrelationreferencesearchwidgetwrapper.cpp:105]: This would be good to add tests for

**Review by nyalldawson — APPROVED** [review]

**Comment by signedav** [comment]
Thanks @nyalldawson

**Pull request merged**